src/services/chris_integration.tsx

```diff
@@ -187,20 +184,22 @@ class ChrisIntegration {
         title: img.fname,
         imagefile: `${filename}.jpg`
       }
+
       if (covidnetPlugin === undefined || covidnetPlugin.data === undefined) {
         return [{
           plugin: pluginNeeded,
           error: new Error('not registered')
         }];
       }
       const covidnetInstance: PluginInstance = await client.createPluginInstance(covidnetPlugin.data.id, plcovidnet_data);
-      console.log("Covidnet Running");
+      console.log(`${pluginNeeded.toUpperCase()} task sent into the task queue`)
+
       const covidnetResult = await pollingBackend(covidnetInstance);
       if (covidnetResult.error) {
         return [covidnetResult];
       }
 
-      return [dirCopyResult, imgConverterResult, covidnetResult];
+      return [covidnetResult];
```

Comment:
We only use the results to check that the analysis was successful by checking that all promises have been settled. Thus, we only care that the last promise has been settled. If things fail in the chain, it would propagate to the last plugin.
```
const promises = [];
    for (let img of dcmImages) {
      promises.push(ChrisIntegration.processOneImg(img));
    }
    const processedImages = await Promise.allSettled(promises);
```

Reply:
Thanks for your question,
It helped me catch another thing that can be improved. Realized that I was returning a 1 item array so made changes to remove the array.